check_addon.py

```diff
@@ -334,6 +339,52 @@ def _check_file_whitelist(error_counter, file_index, addon_path):
     return error_counter
 
 
+def get_repo_addons(branch):
+    branch_url = ROOT_URL.format(branch=branch)
+    content = requests.get(branch_url).content
+    tree = ET.fromstring(content)
+
+    return {
+        a.get("id"): (a.get("version"), bool(a.find("extension/broken")))
+        for a in tree.findall("addon")
+    }
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(error_counter, addon_xml):
+    addon_xml = os.path.join(addon_xml, "addon.xml")
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    current_branch = os.environ.get("TRAVIS_BRANCH")
+
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    for branch in branches:
```

Comment:
Actually you only have to check the previous version.
If it is present in any lower version it will be also present there.